ci/scripts/update-mainnet-artifacts.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# This script is run by the update-mainnet-artifacts job in the ci-pr-only workflow
+# to update ./mainnet-artifacts.bzl bazed on the contents of testnet/mainnet_revisions.json.
+# The latter tracks the IC version (git revision) of the mainnet NNS subnet (with principal tdb26-...).
+#
+# This script will set the MAINNET_NNS_SUBNET_IC_VERSION bazel variable and will compute the SHA256 hashes
+# of the PUBLISHED_BINARIES which are using the MAINNET_NNS_SUBNET_IC_VERSION.
+
+set -eufo pipefail
+
+EXIT_STATUS=0
+
+# Retrieve the IC version (git revision) of the mainnet NNS subnet:
+MAINNET_NNS_SUBNET_ID="tdb26-jop6k-aogll-7ltgs-eruif-6kk7m-qpktf-gdiqx-mxtrf-vb5e6-eqe"
+MAINNET_NNS_SUBNET_IC_VERSION="$(jq ".subnets.\"$MAINNET_NNS_SUBNET_ID\"" -r testnet/mainnet_revisions.json)"
+
+# Write this IC version to mainnet-artifacts.bzl to give bazel access to it:
+sed -i "s/MAINNET_NNS_SUBNET_IC_VERSION = \".*\"/MAINNET_NNS_SUBNET_IC_VERSION = \"$MAINNET_NNS_SUBNET_IC_VERSION\"/" mainnet-artifacts.bzl
+
+# Download the SHA256SUMS file for the binaries of the IC version:
+binaries_SHA256SUMS="$(mktemp -t binaries_SHA256SUMS.XXXX)"
+trap "rm -f $binaries_SHA256SUMS" EXIT
+curl "https://download.dfinity.systems/ic/$MAINNET_NNS_SUBNET_IC_VERSION/binaries/x86_64-linux/SHA256SUMS" \
+    --silent --fail -o "$binaries_SHA256SUMS"
+
+# Construct a sed script that subsitutes the old SHA256 hashes with the new SHA256 hashes
+# found in the previously downloaded SHA256SUMS file.
+# Note that we only update the sha256s of binaries of revision MAINNET_NNS_SUBNET_IC_VERSION.
+# Binaries that track a different revision are left alone. This allows us to temporarily
+# track a different revision for a binary (in case the mainnet binary is incompatible with HEAD in some way)
+# without this script overwriting our override.
+sed_script=""
```

Comment:
why create a sed script instead of repeatedly calling sed? to avoid editing the file while you're reading it?

Reply:
Yes. I was actually doing that before but that failed probably because of reading while writing.